# A/B-тестирование (семинары)
## Урок 5. Применение математической статистики для проверки гипотез в реальной жизни для популярных метрик


### Задача 1. 
Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку.

Выберите метод оценки и оцените есть ли стат.значимые различия между конверсиями в двух группах при alpha = 5%. 

Дайте краткие рекомендации команде.

Результаты:

1)Число юзеров в группах , которые заходили на сайт в период эксперимента: n1 = 15550 и n2 = 15550.

2)Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: n1 = 164 и n2 = 228.

3)Конверсии : conv1 = 1.05% conv2 = 1.47%.


Для оценки статистической значимости различий между конверсиями в двух группах при α=5% можно использовать Z-критерий 
для разности долей. 

Z-критерий находим по формуле Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2)), где p = (n1 * p1 + n2 * p2) / (n1 + n2)

p1 - конверсия в группе 1, 
p2 - конверсия в группе 2,
N1 - количество юзеров в группе 1, которые заходили на сайт в период эксперимента, 
N2 - количество юзеров в группе 2, которые заходили на сайт в период эксперимента, 
n1 и n2 - количество юзеров, которые совершили хотя бы одну покупку за период эксперимента.

In [4]:
from math import sqrt


p1 = 164 / 15550
p2 = 228 / 15550

n1 = 164
n2 = 228

N1, N2 = 15550 (N1 = N2)

p =  (n1 * p1 + n2 * p2) / (n1 + n2)
print('p =',p)

Z = (p1 - p2) / sqrt(p * (1 - p) * (1/N1 + 1/N2))
print('Z =',Z)

p = 0.012940481658901501
Z = -3.2110930329358367


Вывод:

Критическое значение Z-критерия при α=5% и двухсторонней альтернативе равно 1.96.
Т.к. Z меньше критического значения: -3.2110930329358367 < 1.96 , то отвергаем нулевую гипотезу о равенстве конверсий в группах. Т.о., изменения внесенные в форму заказа в магазине Утконос дали статистически значимый эффект. Рекомендуется использовать новую форму заказа.

### Задача 2.

Сравниваем метрику конверсия в покупку. 

Размер выборки - 10000 элементов в каждой группе. 

Какой статистический критерий тут лучше всего подойдёт и почему ?

In [ ]:
Необходимо использовать Z- критерий, т.к. размер выборки достаточно большой, т.е. 10 000 элементов в каждой группе.
Z критерий используется при выборке более 30.